Classes/Domain/AbstractAggregateRoot.php

```diff
@@ -110,7 +104,7 @@ protected function __construct(string $identifier)
      */
     final protected function apply(EventInterface $event)
     {
-        $method = sprintf('when%s', $this->eventTypeService->getEventShortType($event));
+        $method = sprintf('when%s', (new \ReflectionClass($event))->getShortName());
```

Comment:
FYI: I did a quick benchmark and using `ReflectionClass` here doesn't have a noticeable negative effect on performance (over array split-string operation)